src/pages/index.js

```diff
@@ -74,115 +88,266 @@ const closeModalOnEsc = (event) => {
+  closeModal(profileElements.modal);
 });
 
-newPostButton.addEventListener("click", () => {
-  openModal(newPostModal);
+newPostElements.button.addEventListener("click", () => {
+  openModal(newPostElements.modal);
 });
 
-newPostCloseButton.addEventListener("click", () => {
-  closeModal(newPostModal);
+newPostElements.closeButton.addEventListener("click", () => {
+  closeModal(newPostElements.modal);
 });
 
-profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+profileElements.form.addEventListener("submit", handleProfileFormSubmit);
+newPostElements.form.addEventListener("submit", handleNewPostFormSubmit);
+
+cardElements.modalCloseButton.addEventListener("click", () => {
+  closeModal(cardElements.modal);
+});
 
-newPostFormElement.addEventListener("submit", handleNewPostFormSubmit);
+cardElements.modalImage.addEventListener("click", () => {
+  closeModal(cardElements.modal);
+});
 
-// Form Submissions
+// Form Submission Handlers
 function handleProfileFormSubmit(event) {
   event.preventDefault();
 
-  profileNameElement.textContent = profileNameInput.value;
-  profileJobElement.textContent = profileJobInput.value;
+  const submitButton = profileElements.form.querySelector(
+    ".modal__submit-button"
+  );
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
 
-  closeModal(editProfileModal);
+  api
+    .editUserInfo({
+      name: profileElements.nameInput.value,
+      about: profileElements.jobInput.value,
+    })
+    .then((data) => {
+      profileElements.nameElement.textContent = data.name;
+      profileElements.jobElement.textContent = data.about;
+      closeModal(profileElements.modal);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      submitButton.textContent = originalText;
+    });
 }
 
 function handleNewPostFormSubmit(event) {
   event.preventDefault();
 
-  const newCardData = {
-    name: newPostprofileNameInput.value,
-    link: newPostLinkInput.value,
-  };
+  const submitButton = newPostElements.form.querySelector(
+    ".modal__submit-button"
+  );
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
 
-  const cardElement = getCardElement(newCardData);
-  cardList.prepend(cardElement);
+  api
+    .addNewCard({
+      name: newPostElements.nameInput.value,
+      link: newPostElements.linkInput.value,
+    })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardElements.list.prepend(cardElement);
 
-  newPostFormElement.reset();
-  // toggleButtonState(buttonElement);
-  closeModal(newPostModal);
+      newPostElements.form.reset();
+      closeModal(newPostElements.modal);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      submitButton.textContent = originalText;
+      disableSubmitButton(submitButton, validationConfig);
```

Comment:
`disableSubmitButton` should be called in block `then` after a successful response from the server to be able to click it for the 2nd time if there is a server error

**Please, correct it everywhere in the code**